### ISSUE TITLE
SPIKE: Display Next Calendar `event` from Google Cal

### ISSUE THREAD

**Issue by nelsonic** [dwyl/calendar#21]
As a `person` wanting to maximise their personal effectiveness, 
I want to import **`Google Calendar`** data and display it on a single page with other info
so that I don't have to look at _multiple_ places to see what is _next_ in my day.

This [**`SPIKE`**](https://github.com/dwyl/technical-glossary#spike) is scope-limited to displaying my next calendar event on a page.
It should be time-boxed to **`T1d`** (or **`T8h`** split over several sessions). ⏳ 

# Todo

+ [x] In `this` repo, Create a barebones `Phoenix` + `LiveView` App #22 
+ [ ] Should have a landing page explaining the purpose of the single-feature Spike and 
+ [x] a **`Sign-in with Google`** button
  + [x] Use https://github.com/dwyl/elixir-auth-google to request access to a Google Calendar.
  + [x] If there are any modifications required to the scopes/permissions, document them thoroughly
+ [x] Retrieve the list of events for the person who has authenticated
+ [x] Display them on a page _without_ saving to DB.
+ [x] Create a new event and save it back to `Google Calendar` 
  + [x] Read the newly created event from `Google Calendar` and display it in the `Phoenix` App. 

If we can achieve this proof-of-concept we can move forward with more interesting features. 🤞 

# Bonus Level

+ [ ] Research how to store calendar events in `Postgres` DB, document _everything_ you learn! 📝

**Comment by LuchoTurtle** [comment]
@nelsonic Please label a priority so I know when to work on this.

**Comment by nelsonic** [comment]
@LuchoTurtle this is the semi-related side-quest I had mentioned on `Standup` this morning. 
As previously noted, having **`Calendar` integration** is very much on our "roadmap":

<img width="540" alt="image" src="https://user-images.githubusercontent.com/194400/231747423-a8bd1276-319e-4b11-9bc9-0c5f85548971.png">

Essentially I would like to have:
a) a view where I can see the `calendar events` both my own and those of my team.
b) `timers` that I have run automatically create an `event` in my `calendar` 
so that I can _visually_ see how I'm spending my time:

<img width="850" alt="Sample-calendar-view-with-items-and-timers" src="https://user-images.githubusercontent.com/194400/231749063-832ba054-92f8-4abb-a6a6-89ef55bb9156.png">

This is not what the final Look-and-feel will be. This is just a _very_ rough idea.
Basically I want to have a timeline view with the `items` and `timers` visible on my calendar.

**Comment by nelsonic** [comment]
@LuchoTurtle yeah, I was just busy trying to type out what the objective is and was just about to add labels. 👌

**Comment by LuchoTurtle** [comment]
Seems good. 
I have a couple of questions though:

> a) a view where I can see the calendar events both my own and those of my team.
> b) timers that I have run automatically create an event in my calendar
> so that I can visually see how I'm spending my time:

I understand that, to start-off, I want to integrate with Google Calendar and essentially show the person the events they are currently undergoing and have on during the day. 
However, I fail to see how I'll integrate `timers` into this - assuming you're talking about the https://github.com/dwyl/mvp.

**Comment by nelsonic** [comment]
@LuchoTurtle thanks for confirming that this quest is clear. 👌 
For now, we only need to get data in and out of `Google` Calendar. 📆 
If you focus on that we're golden. 🥇 

What I'm describing with creating Calendar `events` out of `items` that have `timers` 
is just a way of recording how I spend my time in a _visual_ way. 
i.e. I want to see on my calendar what I did with a block of time.

In case you're wondering why "Executives" at companies are paid 10-100x what "normal" employees are paid,
it's because the value of their time translates directly to the bottom line. 
i.e. each hour they spend working on a project (or OKR) translates to a fairly easy-to-track dollar amount of **`profit`** for the company. If what you do cannot be tracked then your "value" to the company will be difficult to explain.
Conversely if _everything_ you do is carefully tracked and you can _visualise_ it - first on a calendar, but then subsequently on a pretty graph - then you can take that insight to the `BANK`! 

Once we have the ability to view data from `Google` Calendar we can display it in the `App`. 
e.g: the person's "Next Appointment". 
But that's "nothing new" or terribly exciting. 
Where we want to get to is the ability to answer the question: "How did I spend my day?"
Did I work on my highest priority tasks during the most effective time of the day. 
Did my `P1` task get finished today or did I get distracted by `P2` or `P3` non-mission-critical chores? 

`items` and `timers` are _currently_ content types in the `MVP`, yes!
But if we can persist them to a Calendar as `events` that have a `title` and a `duration` (i.e. `item.text` and `timer.start|stop`) then we can _easily_ see how we spend our time. 
This is gold.

**Comment by nelsonic** [comment]
Please leave updates with where you are stuck getting the data from the `Google Calendar API` 🙏

**Comment by LuchoTurtle** [comment]
I'm going through the errors the API gives me after every change I do but I feel I'm really close to getting proper data from the API. I'm also converting my `app` view from the normal `Controller-View` setup the project had and doing it to `LiveView`, so I need to convert my socket assigns properly.
